subprojects/core/src/main/java/org/gradle/internal/jvm/UnsupportedJavaRuntimeException.java

```diff
@@ -18,15 +18,26 @@
 
 import org.gradle.api.GradleException;
 import org.gradle.api.JavaVersion;
+import org.gradle.util.DeprecationLogger;
 import org.gradle.util.GradleVersion;
 
 public class UnsupportedJavaRuntimeException extends GradleException {
+    public static final String JAVA7_DEPRECATION_WARNING = "Support for running Gradle using Java 7 has been deprecated and is scheduled to be removed in Gradle 5.0. "
```

Comment:
:o: It probably will probably be long until we need to deprecate another Java version, but I think it would be nice to have this more general. So that next time deprecating a Java versions would only require us to change one constant. I am thinking of something like:

```
private static final JavaVersion RECOMMENDED_VERSION = JavaVersion.VERSION_1_8;
    
public static final String JAVA_DEPRECATION_WARNING = "Support for running Gradle using Java" + RECOMMENDED_VERSION.getMajorVersion() +" has been deprecated and is scheduled to be removed in Gradle 5.0. "
        + "Please see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_cross_compilation for more details.";

    private static void javaDeprecationWarning() {
        if (JavaVersion.current().compareTo(RECOMMENDED_VERSION) >= 0) {
            DeprecationLogger.nagUserWith(JAVA_DEPRECATION_WARNING);
        }
    }
```

Reply:
Good catch and good advice, thanks!

Reply:
I think I've fix this, let me wait and see the CI results.